### PR TITLE
feat: add instructions for adding event api to backend

### PR DESCRIPTION
**Pull request by iartemiev**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [x] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [x] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx [hunk 1]
```diff
@@ -89,15 +89,182 @@ export default function App() {
...
 }
 ```
 
-## Connect to an Event API with an existing Amplify backend
+## Add an Event API to an existing Amplify backend
 
-Coming Soon
+This guide walks through how you can add an Event API to an existing Amplify backend.
+We'll be using Cognito User Pools for authenticating with Event API from our frontend application. 
+Any signed in user will be able to subscribe to the Event API and publish events.
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx:105]:
```suggestion
This guide walks through how you can add an Event API to an existing Amplify backend. We'll be using Cognito User Pools for authenticating with Event API from our frontend application. Any signed in user will be able to subscribe to the Event API and publish events.
```

File: src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx [hunk 2]
```diff
@@ -89,15 +89,182 @@ export default function App() {
...
+
+Before you begin, you will need:
+
+- An existing Amplify backend (see [Quickstart](/[platform]/start/quickstart/))
+- Latest versions of `@aws-amplify/backend` and `@aws-amplify/backend-cli` (`npm i @aws-amplify/backend@latest @aws-amplify/backend-cli@latest`)
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx:110]:
```suggestion
- Latest versions of `@aws-amplify/backend` and `@aws-amplify/backend-cli` (`npm add @aws-amplify/backend@latest @aws-amplify/backend-cli@latest`)
```

File: src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx [hunk 3]
```diff
@@ -89,15 +89,182 @@ export default function App() {
...
+    connectionAuthModes: [{ authType: AuthorizationType.USER_POOL }],
+    defaultPublishAuthModes: [{ authType: AuthorizationType.USER_POOL }],
+    defaultSubscribeAuthModes: [{ authType: AuthorizationType.USER_POOL }],
+  },
+});
+
+// create a default namespace for our Event API:
+const namespace = new CfnChannelNamespace(
+  customResources,
+  'cfnEventAPINamespace',
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx:161]:
```suggestion
  'CfnEventAPINamespace',
```

to align with other PascalCase logical IDs

File: src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx [hunk 4]
```diff
@@ -89,15 +89,182 @@ export default function App() {
...
+const backend = defineBackend({
+	auth,
+});
+
+// highlight-start
+// create a new stack for our Event API resources:
+const customResources = backend.createStack('custom-resources');
+
+// add a new Event API to the stack:
+const cfnEventAPI = new CfnApi(customResources, 'cfnEventAPI', {
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx:138]:
```suggestion
const cfnEventAPI = new CfnApi(customResources, 'CfnEventAPI', {
```

File: src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx [hunk 5]
```diff
@@ -89,15 +89,182 @@ export default function App() {
...
+To test your changes, deploy your Amplify Sandbox.
+
+```bash title="Terminal" showLineNumbers={false}
+npx ampx sandbox
+```
+
+### Connect your frontend application
+
+After the sandbox deploys, connect your frontend application to the Event API.
+We'll be using the Amplify Authenticator to sign in to our Cognito User Pool.
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx:208]:
```suggestion
After the sandbox deploys, connect your frontend application to the Event API. We'll be using the [Amplify Authenticator component](https://ui.docs.amplify.aws/react/connected-components/authenticator) to sign in to our Cognito User Pool.
```

File: src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx [hunk 6]
```diff
@@ -89,15 +89,182 @@ export default function App() {
...
+
+If you don't already have the Authenticator installed, you can install it by running `npm i @aws-amplify/ui-react`.
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx:210]:
```suggestion
If you don't already have the Authenticator installed, you can install it by running `npm add @aws-amplify/ui-react`.
```

File: src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx [hunk 7]
```diff
@@ -89,15 +89,182 @@ export default function App() {
...
+
+```tsx title="src/App.tsx"
+import { useEffect, useState } from 'react';
+import { Amplify } from 'aws-amplify';
+import { events, type EventsChannel } from 'aws-amplify/data';
+import { Authenticator } from '@aws-amplify/ui-react';
+import '@aws-amplify/ui-react/styles.css';
+import config from '@/amplify_outputs.json';
+
+Amplify.configure(config);
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx:220]:
```suggestion
import outputs from '@/amplify_outputs.json';

Amplify.configure(outputs);
```

Comment (josefaidt) [reply src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx:220]: minor tweak to use "outputs" language, and to remove any confusion for Vite-based applications we can use a relative import. The `@` alias is available out of the box for Next.js apps, but customers would need to configure this for Vite

**Review by josefaidt — APPROVED** [review]

**Review by stocaaro — APPROVED** [review]

**Pull request merged**